sql/moz-fx-data-shared-prod/firefox_ios_derived/retention_clients_v1/query.sql

```diff
@@ -0,0 +1,65 @@
+WITH clients_daily AS (
+  SELECT
+    submission_date,
+    client_id,
+    sample_id,
+    normalized_channel AS channel,
+    country,
+    first_seen_date,
+    is_new_profile,
+    (app_display_version = '107.2' AND submission_date >= '2023-02-01') AS is_suspicious_device_client,
+  FROM firefox_ios.baseline_clients_last_seen
+  WHERE
+    submission_date = DATE_SUB(@submission_date, INTERVAL 27 DAY)
+),
+
+active_users AS (
+  SELECT
+    submission_date,
+    client_id,
+    -- is_dau,
+    -- is_daily_user,
+    mozfun.bits28.retention(days_seen_bits, submission_date) AS retention_seen,
+    mozfun.bits28.retention(days_seen_bits, submission_date) AS retention_active,
+    isp,
+  FROM firefox_ios.baseline_clients_last_seen
+  -- FROM backfills_staging_derived.active_users -- not using for now because it makes the query way too heavy
+  -- FROM telemetry.active_users
+  WHERE
+    submission_date = @submission_date
+    -- AND app_name LIKE "Firefox iOS%"
+)
+
+SELECT
+  @submission_date AS submission_date,
+  retention_seen.day_27.metric_date AS metric_date,
+  -- clients_daily.submission_date AS metric_date,
+  clients_daily.first_seen_date,
+  client_id,
+  clients_daily.sample_id,
+  clients_daily.channel,
+  clients_daily.country,
+  clients_daily.is_suspicious_device_client,
+  -- clients_daily.is_new_profile AS new_client,
+  -- active_users.app_name,
+  -- active_users ATM too heavy to work with, manually tweaking the app_name
+  CASE
+    WHEN isp = 'BrowserStack'
+      THEN CONCAT('Firefox iOS', ' ', isp)
+    ELSE 'Firefox iOS'
+  END AS app_name,
+  -- ping sent retention
+  active_users.retention_seen.day_27.active_on_metric_date AS day_ping,
+  active_users.retention_seen.day_27.active_on_metric_date AND retention_seen.day_27.active_in_week_3 AS sent_ping_week_4,
+  -- activity retention
+  active_users.retention_active.day_27.active_on_metric_date AS day_active,
+  active_users.retention_active.day_27.active_on_metric_date AND active_users.retention_active.day_27.active_in_week_3 AS retained_week_4,
+  -- new client retention
+  retention_seen.day_27.metric_date = first_seen_date AS is_new_client,
+  retention_seen.day_27.metric_date = first_seen_date AND retention_active.day_27.active_in_week_3 AS new_client_retained_week_4,
+  -- clients_daily.is_new_profile AS new_client,
+  -- clients_daily.is_new_profile AND retention_active.day_27.active_in_week_3 as new_client_retained_week_4
+  -- active_users.retention_seen,
+  -- active_users.retention_active,
+FROM clients_daily
+INNER JOIN active_users USING(client_id)
```

Comment:
We only want to include clients which have sent a ping on the metric date (submission_date - 28 days)